lib/rubygems.rb

```diff
@@ -792,21 +792,18 @@ def self.write_binary(path, data)
   end
 
   ##
-  # Open a file with given flags. It requires special logic on Windows, like
-  # protecting access with flock
+  # Open a file with given flags
 
   def self.open_file(path, flags, &block)
-    if !java_platform? && win_platform?
-      open_file_with_flock(path, flags, &block)
-    else
-      open_file_without_flock(path, flags, &block)
-    end
+    File.open(path, flags, &block)
```

Comment:
This looks that `Gem.read_binary` and `Gem.write_binary` would no longer lock the target file.
Intentional?

Reply:
Yeah. Previously they would only lock the file on Windows, and locking mechanism was not really working properly because depending on the mode, the file would be truncated before being locked. I think it's better to make this method consistent and use a separate method when explicit locking is necessary.

Reply:
If this method doesn't acquire a lock, `Gem.read_binary` probably doesn't need to open in read/write mode.
And `read_binary` and `write_binary` could be simplified by using `File.binread` and `File.binwrite` respectively.

Reply:
PR created at #7932.